homeassistant/components/linkplay/button.py

```diff
@@ -0,0 +1,120 @@
+"""Support for LinkPlay buttons."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+import logging
+from typing import Any, Concatenate
+
+from linkplay.bridge import LinkPlayBridge
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LinkPlayConfigEntry, LinkPlayRequestException
+from .const import DOMAIN
+from .utils import MANUFACTURER_GENERIC, get_info_from_project
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class LinkPlayButtonEntityDescription(ButtonEntityDescription):
+    """Class describing LinkPlay button entities."""
+
+    remote_function: Callable[[LinkPlayBridge], Coroutine[Any, Any, None]]
+
+
+BUTTON_TYPES: tuple[LinkPlayButtonEntityDescription, ...] = (
+    LinkPlayButtonEntityDescription(
+        key="timesync",
+        translation_key="timesync",
+        remote_function=lambda linkplay_bridge: linkplay_bridge.device.timesync(),
+        entity_category=EntityCategory.CONFIG,
+    ),
+    LinkPlayButtonEntityDescription(
+        key="restart",
+        translation_key="restart",
+        remote_function=lambda linkplay_bridge: linkplay_bridge.device.reboot(),
+        entity_category=EntityCategory.CONFIG,
+    ),
```

Comment:
Let's use the device class reboot. You can then remove the name and icon and use the fallback